clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -401,12 +524,12 @@ private void maybeThrowInvalidGroupIdException() {
 
     @Override
     public Map<MetricName, ? extends Metric> metrics() {
-        throw new KafkaException("method not implemented");
+        return Collections.unmodifiableMap(metrics.metrics());
     }
 
     @Override
     public List<PartitionInfo> partitionsFor(String topic) {
-        throw new KafkaException("method not implemented");
+        return partitionsFor(topic, Duration.ofMillis(defaultApiTimeoutMs));
```

Comment:
I had missed a detail here. We're only implementing this with the default timeout, but not the overloaded one below, that has the timeout provided by the user. Is there a reason or is it just that we missed the latter one? (same for the `listTopics`)

Reply:
(I guess it will be all part of the integration with the metadata calls right?)

Reply:
Good call out. I don't know that the necessary topic plumbing code was written at that time. Would you mind filing a bug to resolve?